File: src/main/java/com/devmpv/model/Thread.java
package com.devmpv.model;

import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

/**
 * Thread entity based on {@link Message}
 * 
 * @author devmpv
 *
 */
@Entity
public class Thread extends Message {

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "thread")
    private Set<Message> messages;

    @ManyToOne
    @JoinColumn(name = "board_id", nullable = false)
    private Board board;

    public Thread() {
	setTimestamp(System.currentTimeMillis());
    }

    public Thread(Board board, String title, String text) {
        super(title, text);
        this.board = board;
    }

    public Board getBoard() {
	    return board;
    }

    public Set<Message> getMessages() {
	    return messages;
    }

    public void setBoard(Board board) {
	    this.board = board;
    }

    public void setMessages(Set<Message> messages) {
	    this.messages = messages;
    }
}

File: src/main/kotlin/com/devmpv/model/Thread.kt
package com.devmpv.model

import javax.persistence.*

/**
 * Thread entity based on [Message]
 *
 * @author devmpv
 */
@Entity
class Thread(
        @ManyToOne
        @JoinColumn(name = "board_id", nullable = false)
        var board: Board? = null,

        title: String,

        text: String
) : Message(title, text) {

    @OneToMany(cascade = arrayOf(CascadeType.ALL), fetch = FetchType.LAZY, mappedBy = "thread", targetEntity = Message::class)
    var messages: MutableSet<Message> = HashSet()

    constructor() : this(null, "", "")
}
